src/main/java/coloringpage/Converter.java

```diff
@@ -2,69 +2,113 @@
 
 import javax.imageio.ImageIO;
 import java.awt.*;
-import java.awt.image.BufferedImage;
-import java.awt.image.BufferedImageOp;
-import java.awt.image.ConvolveOp;
-import java.awt.image.Kernel;
+import java.awt.image.*;
 import java.io.File;
 import java.io.IOException;
+import java.util.Arrays;
 
 public class Converter {
     public BufferedImage toLineDrawing(File inputFile) throws IOException {
         BufferedImage image = ImageIO.read(inputFile);
 
+        BufferedImage grayImage = grayscaleImage(image);
+        BufferedImage invertedAndBlurredImage = invertAndBlurImage(grayImage);
+
+        return dodge(invertedAndBlurredImage, grayImage);
+    }
+
+    private BufferedImage grayscaleImage(BufferedImage image) {
+        //https://www.tutorialspoint.com/java_dip/grayscale_conversion.htm
+
         int width = image.getWidth();
         int height = image.getHeight();
 
         for(int i=0; i<height; i++) {
-
             for(int j=0; j<width; j++) {
-
                 Color c = new Color(image.getRGB(j, i));
                 int red = (int)(c.getRed() * 0.299);
                 int green = (int)(c.getGreen() * 0.587);
-                int blue = (int)(c.getBlue() *0.114);
-
-                Color newColor = new Color(red+green+blue,
+                int blue = (int)(c.getBlue() * 0.114);
 
-                        red+green+blue,red+green+blue);
+                int grayscaleColor = red + green + blue;
+                Color newColor = new Color(grayscaleColor, grayscaleColor, grayscaleColor);
 
-                image.setRGB(j,i, (255- newColor.getRGB()));
+                image.setRGB(j,i, newColor.getRGB());
             }
         }
+        return image;
+    }
 
-        //     Resource:   http://www.jhlabs.com/ip/blurring.html
-        float[] matrix = new float[400];
-        for (int i = 0; i < 400; i++) {
-            matrix[i] = 1.0f / 400.0f;
-        }
-
-        int blurRad = 300;
-        float[] blurKernel = new float[blurRad*blurRad];
-        for(int i =0; i<blurRad*blurRad; i++) {
-            blurKernel[i] = 1.0f/256.0f;
-        }
```

Comment:
@Esther sheina, it looked like this code didn't do anything, so I deleted it, confirm?

Reply:
NOOO!!!!
That's the blur function!!! That's how you set the blurriness!

Reply:
so thats what I thought, but then the function `blurImage()` (which I renamed from `processImage()`) does that as well, and this chuck of code isn't used